testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/BrowserFlowTest.java

```diff
@@ -190,10 +214,258 @@ public void userWithTwoAdditionalFactors() throws InterruptedException {
+                        )
+                )
+                .defineAsBrowserFlow()
+        );
+    }
+
+    // Check the ConditionalBlockRoleAuthenticator
+    // Configure a conditional subflow with the required role "user" and an OTP authenticator
+    // user-with-two-configured-otp has the "user" role and should be asked for an OTP code
+    // user-with-one-configured-otp does not have the role. He should not be asked for an OTP code
+    @Test
+    public void testConditionalBlockRoleAuthenticator() {
+        String requiredRole = "user";
+        // A browser flow is configured with an OTPForm for users having the role "user"
+        configureBrowserFlowOTPNeedsRole(requiredRole);
+
+        try {
+            // user-with-two-configured-otp has been configured with role "user". He should be asked for an OTP code
+            provideUsernamePassword("user-with-two-configured-otp");
+            Assert.assertTrue(oneTimeCodePage.isOtpLabelPresent());
+            loginTotpPage.assertCurrent();
+            loginTotpPage.assertCredentialsComboboxAvailability(true);
+
+            // user-with-one-configured-otp has not configured role. He should not be asked for an OTP code
+            provideUsernamePassword("user-with-one-configured-otp");
+            Assert.assertFalse(oneTimeCodePage.isOtpLabelPresent());
+            Assert.assertFalse(loginTotpPage.isCurrent());
+        } finally {
+            testingClient.server("test").run(setBrowserFlowToRealm());
+        }
+    }
+
+    // Configure a flow with a conditional sub flow with a condition where a specific role is required 
+    private void configureBrowserFlowOTPNeedsRole(String requiredRole) {
+        final String newFlowAlias = "browser - rule";
+        testingClient.server("test").run(session -> FlowUtil.inCurrentRealm(session).copyBrowserFlow(newFlowAlias));
+        testingClient.server("test").run(session -> FlowUtil.inCurrentRealm(session)
+                .selectFlow(newFlowAlias)
+                .inForms(forms -> forms
+                        .clear()
+                        // Update the browser forms with a UsernamePasswordForm
+                        .addAuthenticatorExecution(Requirement.REQUIRED, UsernamePasswordFormFactory.PROVIDER_ID)
+                        .addSubFlowExecution(Requirement.CONDITIONAL, subFlow -> subFlow
+                                .addAuthenticatorExecution(Requirement.REQUIRED, ConditionalBlockRoleAuthenticatorFactory.PROVIDER_ID,
+                                        config -> config.getConfig().put("condUserRole", requiredRole))
+                                .addAuthenticatorExecution(Requirement.REQUIRED, OTPFormAuthenticatorFactory.PROVIDER_ID)
+                        )
+                )
+                .defineAsBrowserFlow()
+        );
+    }
+
+    @Test
+    public void testAlternativeSubFlowWithAlwaysValidatingExecutor() {
```

Comment:
Is this test incomplete?

Reply:
Right... I started working on it but had to switch on something else... I remove it for now.